pkg/distributor/distributor.go

```diff
@@ -534,15 +563,7 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 				continue
 			}
 
-			shardStreamsCfg := d.validator.Limits.ShardStreams(tenantID)
-			if shardStreamsCfg.Enabled {
-				streams = append(streams, d.shardStream(stream, pushSize, tenantID)...)
-			} else {
-				streams = append(streams, KeyedStream{
-					HashKey: lokiring.TokenFor(tenantID, stream.Labels),
-					Stream:  stream,
-				})
-			}
+			maybeShardByTime(stream, lbs, pushSize)
```

Comment:
I would call this function `shardStreams`, which would do sharding by the time (if enabled) and sharding by the rate(if enabled).

Reply:
:+1: but I will rename it to `maybeShardStreams` because the `if` checks are in the function itself

Reply:
Resolved by https://github.com/grafana/loki/pull/14711/commits/ad1c0b2ad882121da20acda5e50db3833ac0a39d